internal/encryption/keyring/ring.go

```diff
@@ -23,6 +23,15 @@ func Default() Ring {
 	return defaultRing
 }
 
+// SetDefault overrides the default keyring.
+// Note: This function is defined for testing purpose.
+// Use Init to correctly setup a keyring.
+func SetDefault(r Ring) {
```

Comment:
If only for testing, I think `Mock` makes things clear.

```suggestion
func MockDefault(r Ring) {
```